app/celery/process_sms_client_response_tasks.py

```diff
@@ -36,9 +36,10 @@ def process_sms_client_response(self, status, provider_reference, client_name, c
 
     # validate status
     try:
-        notification_status = response_parser(status)
+        notification_status, detailed_status = response_parser(status, code)
```

Comment:
Optional: We have terminology `code`, `substatus`, `reason`, `detailed_status`. Worth thinking about if these could or should be more consistent. It may not be appropriate though so give it some thought